app/models/gem_dependent.rb

```diff
@@ -0,0 +1,56 @@
+class GemDependent
+  attr_reader :gem_names
+
+  def initialize(gem_names)
+    @gem_information = {}
+    @gem_names = gem_names
+  end
+
+  def fetch_dependencies
+    dependencies = []
+    @gem_names.each { |g| @gem_information[g] = "deps/v1/#{g}" }
+
+    @gem_information.each do |gem_name, cache_key|
+      if memcached_gem_info.keys.include?(cache_key)
+        # Fetch the gem's dependencies from the cache
+        StatsD.increment 'gem_dependent.memcached.hit'
+        dependencies << memcached_gem_info[cache_key]
+      else
+        # Fetch the gem's dependencies from the database
+        StatsD.increment 'gem_dependent.memcached.miss'
+        result = fetch_dependency_from_db(gem_name)
+        memcached_client.write(cache_key, result)
+        dependencies << result
+      end
+    end
+
+    dependencies
+  end
+
+  alias to_a fetch_dependencies
+
+  private
+
+  def fetch_dependency_from_db(gem_name)
+    gem_record = Rubygem.includes(:versions).find_by_name!(gem_name)
+    gem_record.versions.includes(:dependencies).sort_by(&:number).reverse_each.map do |version|
+      version_deps = version.dependencies.where(scope: 'runtime').all
```

Comment:
Turns out the `where` in this line causes an additional query. I think we can change this line to: `version_deps = version.dependencies.select { |d| d.scope == 'runtime' }`